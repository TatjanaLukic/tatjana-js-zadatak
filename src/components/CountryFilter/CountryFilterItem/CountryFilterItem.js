import React from "react";
import PropTypes from "prop-types";

const CountryFilterItem = ({
  id,
  value,
  checked,
  onChange,
  labelName,
  className,
}) => {
  return (
    <div>
      <input
        type="radio"
        id={id}
        value={value}
        onChange={onChange}
        checked={checked}
        hidden
      />
      <label
        htmlFor={id}
        className={checked ? `${className} ${className}--checked` : className}
      >
        {labelName}
      </label>
    </div>
  );
};

CountryFilterItem.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  labelName: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default CountryFilterItem;
