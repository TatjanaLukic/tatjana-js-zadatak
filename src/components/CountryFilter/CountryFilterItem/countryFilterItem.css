.country-filter-item {
  background-color: white;
  border: none;
  color: #ee10ee;
  text-align: center;
  padding: 0.3rem;
  margin-right: 1.7rem;
}
@media screen and (min-width: 576px) {
  .country-filter-item {
    margin: 0.5rem;
    font-weight: bold;
    width: 30px;
    height: 30px;
  }
}

.country-filter-item--checked {
  color: white;
  background-color: #8e0a8e;
}

.country-filter-item--disabled {
  pointer-events: none;
  opacity: 0.3;
}

/*# sourceMappingURL=countryFilterItem.css.map */
