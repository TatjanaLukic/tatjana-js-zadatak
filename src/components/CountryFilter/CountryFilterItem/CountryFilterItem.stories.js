import React from "react";
import CountryFilterItem from "./CountryFilterItem";

export default {
  title: "App/CountryFilterItem",
  component: CountryFilterItem,
};

const onFn = () => {};

const Template = (args) => <CountryFilterItem {...args} />;

export const Default = Template.bind({});
Default.args = {
  id: "us",
  value: "us",
  onChange: { onFn },
  checked: false,
  disabled: false,
  labelName: "US",
  className: "country-filter-item",
};

export const DefaultDisabled = Template.bind({});
DefaultDisabled.args = {
  ...Default.args,
  disabled: true,
};

export const Checked = Template.bind({});
Checked.args = {
  ...Default.args,
  checked: true,
};

export const CheckedDisabled = Template.bind({});
CheckedDisabled.args = {
  ...Default.args,
  checked: true,
  disabled: true,
};
