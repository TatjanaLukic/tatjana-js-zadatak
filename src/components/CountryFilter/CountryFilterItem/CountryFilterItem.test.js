import React from "react";
import { render, screen } from "@testing-library/react";
import CountryFilterItem from "./CountryFilterItem";

describe("<CountryFilterItem />", () => {
  let countryFilterItem;
  let fn = () => {};
  let data = {
    id: "us",
    value: "us",
    checked: false,
    onChange: fn,
    labelName: "us",
    className: "country-filter-item",
  };

  it("should render radio with hidden input and visible label", () => {
    render(<CountryFilterItem {...data} />);
    const countryFilterItem = screen.getByTestId("cf-item");

    expect(
      countryFilterItem.querySelector("input[type=radio]")
    ).toBeInTheDocument();
    expect(
      countryFilterItem.querySelector("input[type=radio]")
    ).toHaveAttribute("hidden");

    expect(countryFilterItem.querySelector("label")).toBeInTheDocument();
    expect(countryFilterItem.querySelector("label").textContent).toBe("us");
    expect(countryFilterItem.querySelector("label")).not.toHaveAttribute(
      "hidden"
    );
  });

  it("matches snapshoot ", async () => {
    const { container } = render(<CountryFilterItem {...data} />);
    expect(container).toMatchSnapshot();
  });
});
