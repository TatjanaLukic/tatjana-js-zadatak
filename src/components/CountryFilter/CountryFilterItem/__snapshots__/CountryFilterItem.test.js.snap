// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryFilterItem /> matches snapshoot  1`] = `
<div>
  <div
    data-testid="cf-item"
  >
    <input
      hidden=""
      id="us"
      type="radio"
      value="us"
    />
    <label
      class="country-filter-item"
      for="us"
    >
      us
    </label>
  </div>
</div>
`;
